src/hooks/hookResources.ts

```diff
@@ -503,6 +503,154 @@ export function handleHook(
+                  const reference = e?.resource;
+                  const request = entry.resource as MedicationRequest;
+
+                  // add the reference as a contained resource to the request
+                  if (!request?.contained) {
+                    request.contained = [];
+                    request.contained.push(reference);
+                  } else {
+                    // only add to contained if not already in there
+                    let found = false;
+                    request.contained.forEach(c => {
+                      if (c.id === reference.id) {
+                        found = true;
+                      }
+                    });
+                    if (!found) {
+                      request.contained.push(reference);
+                    }
+                  }
+                }
+              }
+            }
+          }
+        });
+      }
+    }
+  });
+}
+
+// handles order-sign and order-select currently
+export async function handleCardEncounter(
+  res: any,
+  hookPrefetch: HookPrefetch | undefined,
+  contextRequest: FhirResource | undefined,
+  patient: FhirResource | undefined
+) {
+  //TODO: should we add the other pdf information links to the card, or just have the smart links?
+
+  const medResource = hookPrefetch?.medicationRequests;
+  const medicationRequestsBundle = medResource?.resourceType === 'Bundle' ? medResource : undefined;
+
+  // create empty card array
+  const cardArray: Card[] = [];
+
+  // find all matching rems cases for the patient
+  const patientName = patient?.resourceType === 'Patient' ? patient?.name?.[0] : undefined;
+  const patientBirth = patient?.resourceType === 'Patient' ? patient?.birthDate : undefined;
+  const remsCaseList = await remsCaseCollection.find({
+    patientFirstName: patientName?.given?.[0],
+    patientLastName: patientName?.family,
+    patientDOB: patientBirth
+  });
+
+  // loop through all the rems cases in the list
+  for (const remsCase of remsCaseList) {
+    // find the drug in the medicationCollection that matches the REMS case to get the smart links
+    const drug = await medicationCollection
+      .findOne({
+        code: remsCase.drugCode,
+        name: remsCase.drugName
+      })
+      .exec();
+
+    // get the rule summary from the codemap
+    const codeRule = codeMap[remsCase.drugCode];
+    let summary = '';
+    for (const rule of codeRule) {
+      if (rule.stakeholderType === 'patient') {
+        summary = rule.summary || remsCase.drugName || 'Rems';
+      }
+    }
+
+    // create the card
+    let smartLinkCount = 0;
+    const card = new Card(summary, CARD_DETAILS, source, 'info');
+
+    // process the MedicationRequests to add the Medication into contained resources
+    if (medicationRequestsBundle) {
+      processMedicationRequests(medicationRequestsBundle);
+    }
+
+    // find the matching MedicationRequest for the context
+    const request = medicationRequestsBundle?.entry?.filter(entry => {
```

Comment:
If we're only looking for one BundleEntry (and not a list) with a particular resource (in this case, the first) then doing a filter isn't necessary. 
```suggestion
    const request = medicationRequestsBundle?.entry?.find(entry => {
```